numpy/core/tests/test_datetime.py

```diff
@@ -20,6 +21,14 @@
     _has_pytz = False
 
 
+@contextlib.contextmanager
+def assert_deprecated():
```

Comment:
No, but this isn't correct, `w` is empty if the test fails and you will get an `IndexError`. We should fix up `assert_warns` so that it works. The problem (I think) is that `DeprecationWarning` and a few others are turned into errors for development testing. We usually do something like
```
    with warnings.catch_warnings():
        warnings.simplefilter('error', DeprecationWarning)
        assert_raises(DeprecationWarning, ...)
```
but that is clumsy. Let me take a look at `assert_warns`.